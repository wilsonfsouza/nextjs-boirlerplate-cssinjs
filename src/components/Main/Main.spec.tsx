import { screen, render } from '@testing-library/react';

import { Main } from '.';

describe('<Main />', () => {
  it('should render the heading', () => {
    const { container } = render(<Main />);

    const heading = screen.getByRole('heading', { name: /advanced react/i });

    expect(heading).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the colors correctly', () => {
    const { container } = render(<Main />);

    expect(container.firstChild).toHaveStyle({ 'background-color': '#06092b' });
  });
});
