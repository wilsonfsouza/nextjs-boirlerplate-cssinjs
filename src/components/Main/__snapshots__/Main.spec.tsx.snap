// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
<main
  class="styles__Wrapper-sc-dzx1ao-0 laXFAU"
>
  <img
    alt="Image of an atom and Advanced React written side by side"
    class="styles__Logo-sc-dzx1ao-1 fqmfdP"
    src="/assets/images/logo.svg"
  />
  <h1
    class="styles__Heading-sc-dzx1ao-2 dHzhqJ"
  >
    Advanced React
  </h1>
  <h3
    class="styles__SubHeading-sc-dzx1ao-3 gbuutk"
  >
    TypeScript, React, NextJS, and Styled Components
  </h3>
  <img
    alt="A developer programming in front of a computer screen"
    class="styles__Illustration-sc-dzx1ao-4 dZrGPP"
    src="/assets/images/hero-illustration.svg"
  />
</main>
`;
